pkg/selectors/kernel.go

```diff
@@ -639,16 +640,21 @@ func parseMatchCapabilityChanges(k *KernelSelectorState, actions []v1alpha1.Capa
 	return nil
 }
 
-func parseMatchBinary(k *KernelSelectorState, index uint32, b *v1alpha1.BinarySelector) error {
+func parseMatchBinary(k *KernelSelectorState, b *v1alpha1.BinarySelector) error {
 	op, err := selectorOp(b.Operator)
 	if err != nil {
 		return fmt.Errorf("matchpid error: %w", err)
 	}
 	WriteSelectorUint32(k, op)
-	WriteSelectorUint32(k, index)
-	WriteSelectorUint32(k, index)
-	WriteSelectorUint32(k, index)
-	WriteSelectorUint32(k, index)
+	if len(b.Values) > 4 {
+		return fmt.Errorf("Only support up to 4 values in MatchBinary")
+	}
+	for _, s := range b.Values {
+		WriteSelectorUint32(k, k.AddBinaryName(s))
+	}
+	for l := len(b.Values); l < 4; l++ {
+		WriteSelectorUint32(k, math.MaxUint32)
+	}
```

Comment:
aren't these loops the same or could be one loop?

Reply:
Here, we should call `WriteSelectorUint32` 4 times exactly.  The number of iterations in the first loop is equal to the number of binary names that the user provides (always <= 4). In the case where the first loop does < 4 iterations, the second loop will fill the empty entries with `math.MaxUint32`.

Reply:
Alright got you! so you can improve later those hardcoded 4 with their corresponding consts to make it easy